airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java

```diff
@@ -96,61 +98,86 @@ private void submitJob(Job job) {
+  private ImmutableMap.Builder<String, Object> generateMetadata(Job job) {
+    final Builder<String, Object> metadata = ImmutableMap.builder();
+    metadata.put("job_type", job.getConfig().getConfigType());
+    metadata.put("job_id", job.getId());
+    metadata.put("attempt_id", job.getAttempts());
+    // build a deterministic job and attempt uuids based off of the scope,which should be unique across
+    // all instances of airbyte installed everywhere).
+    final UUID jobUuid = UUID.nameUUIDFromBytes((job.getScope() + job.getId() + job.getAttempts()).getBytes(Charsets.UTF_8));
+    final UUID attemptUuid = UUID.nameUUIDFromBytes((job.getScope() + job.getId() + job.getAttempts()).getBytes(Charsets.UTF_8));
+    metadata.put("jobUuid", jobUuid);
+    metadata.put("attempt_uuid", attemptUuid);
+
+    switch (job.getConfig().getConfigType()) {
+      case CHECK_CONNECTION_SOURCE, DISCOVER_SCHEMA -> {
+        final StandardSourceDefinition sourceDefinition = configRepository
+            .getSourceDefinitionFromSource(UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+
+        metadata.put("source_definition_name", sourceDefinition.getName());
+        metadata.put("source_definition_id", sourceDefinition.getSourceDefinitionId());
+      }
+      case CHECK_CONNECTION_DESTINATION -> {
+        final StandardDestinationDefinition destinationDefinition = configRepository
+            .getDestinationDefinitionFromDestination(UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+
+        metadata.put("destination_definition_name", destinationDefinition.getName());
+        metadata.put("destination_definition_id", destinationDefinition.getDestinationDefinitionId());
+      }
+      case GET_SPEC -> {
+        // no op because this will be noisy as heck.
+      }
+      case SYNC -> {
+        final StandardSourceDefinition sourceDefinition = configRepository
+            .getSourceDefinitionFromConnection(UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+        final StandardDestinationDefinition destinationDefinition = configRepository
+            .getDestinationDefinitionFromConnection(UUID.fromString(ScopeHelper.getConfigId(job.getScope())));
+
+        metadata.put("connection_id", ScopeHelper.getConfigId(job.getScope()));
+        metadata.put("source_definition_name", sourceDefinition.getName());
+        metadata.put("source_definition_id", sourceDefinition.getSourceDefinitionId());
+        metadata.put("destination_definition_name", destinationDefinition.getName());
+        metadata.put("destination_definition_id", destinationDefinition.getDestinationDefinitionId());
+      }
     }
+    return metadata;
+  }
+
+  private static JobStatus getStatus(OutputAndStatus<?> output) {
+    return switch (output.getStatus()) {
+      case SUCCEEDED -> JobStatus.COMPLETED;
+      case FAILED -> JobStatus.FAILED;
+      default -> throw new IllegalStateException("Unknown state " + output.getStatus());
```

Comment:
This should probably log instead of throw an exception

Reply:
why?

Reply:
mb -- I misread. This is correct.

Reply:
I thoguth this was only used in tracking code not in business logic. Wouldn't have wanted to fail over tracking issues.